src/flask_wtf/file.py

```diff
@@ -22,23 +19,6 @@ def process_formdata(self, valuelist):
         else:
             self.raw_data = ()
 
-    def has_file(self):
-        """Return ``True`` if ``self.data`` is a
-        :class:`~werkzeug.datastructures.FileStorage` object.
-
-        .. deprecated:: 0.14.1
-            ``data`` is no longer set if the input is not a non-empty
-            ``FileStorage``. Check ``form.data is not None`` instead.
-        """
-
-        warnings.warn(
-            FlaskWTFDeprecationWarning(
-                '"has_file" is deprecated and will be removed in 1.0. The data is '
-                "checked during processing instead."
-            )
-        )
-        return bool(self.data)
```

Comment:
@azmeuk / @davidism  So how does a user check whether the file field has data? The deprecation is a triple negative and I find it hard to understand `'data' is no longer set if the input is not a non-empty"`. That triple negative is hard to understand.

The bit in the documentation I can find from the change log https://flask-wtf.readthedocs.io/en/1.0.x/changes/#version-0-14-1 says:

> "has_file is deprecated; it’s now equivalent to bool(field.data)."

As that's the only mention in the change log I can see, I presume I can write `bool(field.data)` still? Which sort of makes me wonder why remove the `has_file()` method?